File: ch2/old/akkademy-db-scala/src/main/scala/com/akkademy/messages.scala
package com.akkademy

package object messages {
  sealed trait Request
  case class Get(key: String) extends Request
  case class Set(key: String, value: String) extends Request
  case class SetIfNotExists(key: String, value: Object) extends Request
  case class Delete(key: String) extends Request

  trait Response
  trait Ok extends Response
  trait NotOk extends Response
}

File: ch2/old/akkademy-db-scala/src/main/scala/com/akkademy/Messages.java
package com.akkademy;

public interface Messages {
    public static interface Request {}
    public static class Get implements Request {
        public final String key;

        public Get(String key) {
            this.key = key;
        }
    }

    public static class Set implements Request {
        public final String key;
        public final Object value;

        public Set(String key, Object value) {
            this.key = key;
            this.value = value;
        }
    }

    public static class SetIfNotExists implements Request {
        public final String key;
        public final Object value;

        public SetIfNotExists(String key, Object value) {
            this.key = key;
            this.value = value;
        }
    }

    public static class Delete implements Request {
        public final String key;

        public Delete(String key) {
            this.key = key;
        }
    }

    public static interface Response {}
    public static class Ok implements Response {}
    public static class Faiure implements Response {}
}
